oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,92 @@ impl super::Node for WasmNode {
     }
 }
 
-#[test]
-fn wasm_invalid_args() {
-    use super::Node;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::node::Node;
     use crate::runtime::TEST_NODE_ID;
-    let configuration = crate::runtime::Configuration {
-        nodes: HashMap::new(),
-        entry_module: "test_module".to_string(),
-        entrypoint: "test_function".to_string(),
-    };
-    let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
-    let (_, reader_handle) =
-        runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
-
-    // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
-    // Minimal module:
-    // \0asm - magic
-    // 0x01 - version (in little-endian)
-    let module =
-        wasmi::Module::from_buffer(vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]).unwrap();
-
-    let mut node = WasmNode::new(
-        "test",
-        runtime_ref,
-        TEST_NODE_ID,
-        Arc::new(module),
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use std::env;
+    use wat::parse_file;
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
+        // Minimal module:
+        // \0asm - magic
+        // 0x01 - version (in little-endian)
+        let module = wasmi::Module::from_buffer(buffer).unwrap();
+
+        Box::new(WasmNode::new(
+            "test",
+            runtime_ref,
+            TEST_NODE_ID,
+            Arc::new(module),
+            entrypoint.to_string(),
+            reader_handle,
+        ))
+    }
+
+    fn from_testfile(file_name: &str) -> Vec<u8> {
+        let mut path = env::current_dir().unwrap();
+        path.pop();
+        path.pop();
+        path.push("testdata");
+        path.push(file_name);
+        parse_file(path).unwrap()
+    }
+
+    #[test]
+    fn wasm_with_invalid_args_fails() {
+        let buffer = vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00];
+
+        // Wrong magic: The minimal module does not have the necessary entry point, so it should
```

Comment:
Is this because of wrong magic or because of missing entry point? AFAICT the magic number is fine here.

Just to make sure we are on the same page, I mean this magic number: https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#high-level-structure

Reply:
You are right. I just noticed that this test is the same as `wasm_starting_module_without_content_fails`, so I've merged the two (sort of)!